src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32.cs

```diff
@@ -358,8 +449,9 @@ private static ParallelQuery<Win32> StartMenuPrograms(string[] suffixes)
 
             var programs1 = paths.AsParallel().Where(p => Extension(p) == ShortcutExtension).Select(LnkProgram);
             var programs2 = paths.AsParallel().Where(p => Extension(p) == ApplicationReferenceExtension).Select(Win32Program);
+            var programs3 = paths.AsParallel().Where(p => Extension(p) == InternetShortcutExtension).Select(InternetShortcutProgram);
 
-            return programs1.Concat(programs2).Where(p => p.Valid);
+            return programs1.Concat(programs2).Where(p => p.Valid).Concat(programs3).Where(p => p.Valid);
```

Comment:
A couple nit points. 
1.  String.Equals should be used over == for comparing strings.  I would use OrdinalIgnoreCase here.
2.  It would be more performant to filter `Where(p=> p.Valid)' first so you only invoke the string comparison on Valid programs.

Reply:
2. Will do that. Thanks for pointing that out.
1. There are a lot of places where that change needs to be made. Do I do it in all places as a part of this PR or do you suggest making another PR with only those changes?

Reply:
@ryanbodrug-microsoft Regarding 2, the code flow is such that it gets all the paths from the start menu, filters those which are say shortcuts ( by performing string comparison so it filters all those paths that have an lnk extension). Then the LnkProgram() function is called on these paths which returns application objects with all the metadata in them. The app being valid or not is one of them. So I don't think we can check that the app is valid before actually performing the string comparison and creating the app. Let me know your thoughts on the same.